docs/Quick_start_guide.rst

```diff
@@ -65,46 +90,147 @@ Types can also be obtained from their name in the C programming language:
+
+  image = itk.imread('image.tif')
+
+  # Read in with a specific pixel type.
+  image = itk.imread('image.tif', itk.F)
+
+  # Read in an image series.
+  # Pass a sorted list of files.
+  image = itk.imread(['image1.png', 'image2.png', 'image3.png'])
+
+  # Read in a volume from a DICOM series.
+  # Pass a directory.
+  # Only a single series, sorted spatially, will be returned.
+  image = itk.imread('/a/dicom/directory/')
+
+  # Write an image.
+  itk.imwrite(image, 'image.tif')
+
+
+  # Read a mesh.
+  mesh = itk.meshread('mesh.vtk')
+
+  # Write a mesh.
+  itk.meshwrite(mesh, 'mesh.vtk')
+
+
+  # Read a spatial transform.
+  transform = itk.transformread('transform.h5')
+
+  # Write a spatial transform.
+  itk.transformwrite(transform, 'transform.h5')
+
+Image filters and Image-like inputs and outputs
+...............................................
+
+All `itk` functional image filters operate on an `itk.Image` but also:
+
+- `xarray.DataArray <https://xarray.pydata.org/en/stable/generated/xarray.DataArray.html>`_ *
+- `numpy.ndarray <https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html>`_
+- `dask.array.Array <https://docs.dask.org/en/latest/array.html>`_
+
+* Preserves image metadata
+
+Filter parameters
 .................
 
 ITK filter parameters can be specified in the following ways:
 
 .. literalinclude:: code/FilterParameters.py
    :lines: 10-
 
+Filter types
+............
 
-Mixing ITK and NumPy
---------------------
+In `itk`, filters are optimized at compile time for each image pixel type and
+image dimension. There are two ways to instantiate these filters with the `itk`
+Python wrapping:
 
-A common use case for using ITK in Python is to mingle NumPy and ITK operations on raster data. ITK provides a large number of I/O image formats and several sophisticated image processing algorithms not available in any other packages. The ability to intersperse that with the SciPy ecosystem provides a great tool for rapid prototyping.
+- *Implicit (recommended)*: Type information is automatically detected from the data. Typed filter objects and images are implicitly created.
 
-The following script shows how to integrate NumPy and ITK:
+.. literalinclude:: code/ImplicitInstantiation.py
+   :lines: 8-
 
-.. literalinclude:: code/MixingITKAndNumPy.py
-   :lines: 8-33
+- *Explicit*: This can be useful if an appropriate type cannot be determined implicitly or when a different filter type than the default is desired.
 
+To specify the type of the filter, use the `ttype` keyword argument. Explicit instantiation of a median image filter:
 
-Similar functions are available to work with `itk.Matrix`, VNL vectors and matrices:
+.. literalinclude:: code/ExplicitInstantiation.py
+   :lines: 8-
 
-.. literalinclude:: code/MixingITKAndNumPy.py
-   :lines: 35-
+Instantiate an ITK object
+.........................
+
+There are two types of ITK objects. Most ITK objects (images, filters, adapters, ...) are instantiated the following way:
```

Comment:
I'd avoid using the ellipsis. If there is not an easy way to list all objects, I'd put it as "Most ITK objects (such as images, filters, or adapters, among others) are  (...)", or "Most ITK objects (such as images, filters, or adapters, for example) are  (...)" or simply "Most ITK objects (such as images, filters, or adapters) are  (...)"

Reply:
:+1: